components/org.wso2.carbon.identity.workflow.engine/src/main/java/org/wso2/carbon/identity/workflow/engine/internal/dao/impl/ApprovalTaskDAOImpl.java

```diff
@@ -503,6 +504,46 @@ public void deletePendingApprovalTasks(String workflowRequestId) throws Workflow
         }
     }
 
+    @Override
+    public List<String> getPendingRequestsByWorkflowId(String workflowId) throws WorkflowEngineServerException {
+
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        try {
+            return jdbcTemplate.executeQuery(WorkflowEngineConstants.SqlQueries.
+                            GET_ALL_UNFINISHED_APPROVAL_EVENTS_BY_WORKFLOW_ID, (resultSet, rowNumber) ->
+                            resultSet.getString(WorkflowEngineConstants.REQUEST_ID_COLUMN),
+                    preparedStatement -> preparedStatement.setString(1, workflowId));
+        } catch (DataAccessException e) {
+            String errorMessage = String.format("Error occurred while retrieving approval tasks for the " +
+                    "workflow id: %s", workflowId);
+            log.debug(errorMessage, e);
+            throw new WorkflowEngineServerException(errorMessage, e);
+        }
+    }
+
+    @Override
+    public List<ApprovalTaskRelationDTO> getApprovalTaskRelationsByRequestId(String requestId)
+            throws WorkflowEngineServerException {
+
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        try {
+            return jdbcTemplate.executeQuery(WorkflowEngineConstants.SqlQueries.
+                            GET_APPROVAL_TASK_RELATIONS_BY_REQUEST_ID, (resultSet, rowNumber) -> {
+                            ApprovalTaskRelationDTO dto = new ApprovalTaskRelationDTO();
```

Comment:
Can we rename this to a meaningful variable name?
```suggestion
                            ApprovalTaskRelationDTO approvalTaskRelationDTO = new ApprovalTaskRelationDTO();
```